File: src/LiteGraph.Server/Classes/ChatEndpointPreloadResult.cs
namespace LiteGraph.Server.Classes
{
    using System;
    using LiteGraph;

    /// <summary>
    /// Chat endpoint model preload result.  Preloading warms the configured model on the upstream
    /// inference endpoint so the first completion does not pay the model load cost.
    /// </summary>
    public class ChatEndpointPreloadResult
    {
        #region Public-Members

        /// <summary>
        /// Chat endpoint GUID.
        /// </summary>
        public Guid EndpointGUID { get; set; } = Guid.Empty;

        /// <summary>
        /// Model configured on the endpoint.
        /// </summary>
        public string Model { get; set; } = null;

        /// <summary>
        /// Provider type of the endpoint.
        /// </summary>
        public ChatProviderTypeEnum Provider { get; set; } = ChatProviderTypeEnum.OpenAI;

        /// <summary>
        /// Whether the provider supports model preloading.  False for cloud providers, whose models are always resident.
        /// </summary>
        public bool Supported { get; set; } = false;

        /// <summary>
        /// Whether a background warm-up was started by this request.
        /// </summary>
        public bool Started { get; set; } = false;

        /// <summary>
        /// Whether a warm-up for this endpoint was already in flight.
        /// </summary>
        public bool AlreadyInProgress { get; set; } = false;

        #endregion

        #region Private-Members

        #endregion

        #region Constructors-and-Factories

        /// <summary>
        /// Instantiate.
        /// </summary>
        public ChatEndpointPreloadResult()
        {

        }

        #endregion

        #region Public-Methods

        #endregion

        #region Private-Methods

        #endregion
    }
}

File: sdk/js/src/models/ChatEndpointPreloadResult.js
/**
 * ChatEndpointPreloadResult class representing the result of a chat endpoint model preload (warm-up) request.
 */
export default class ChatEndpointPreloadResult {
  /**
   * @param {Object} result - Information about the preload result.
   * @param {string|null} [result.EndpointGUID=null] - Chat endpoint GUID (default is null).
   * @param {string|null} [result.Model=null] - Model configured on the endpoint (default is null).
   * @param {string} [result.Provider='OpenAI'] - Provider type of the endpoint (default is 'OpenAI').
   * @param {boolean} [result.Supported=false] - Indicates whether the provider supports model preloading (default is false).
   * @param {boolean} [result.Started=false] - Indicates whether a background warm-up was started (default is false).
   * @param {boolean} [result.AlreadyInProgress=false] - Indicates whether a warm-up was already in flight (default is false).
   */
  constructor(result = {}) {
    const {
      EndpointGUID = null,
      Model = null,
      Provider = 'OpenAI',
      Supported = false,
      Started = false,
      AlreadyInProgress = false,
    } = result;

    this.EndpointGUID = EndpointGUID; // Chat endpoint GUID
    this.Model = Model; // Model configured on the endpoint
    this.Provider = Provider; // Provider type of the endpoint
    this.Supported = Supported; // Indicates if the provider supports preloading
    this.Started = Started; // Indicates if a warm-up was started
    this.AlreadyInProgress = AlreadyInProgress; // Indicates if a warm-up was already in flight
  }
}
